### DISCUSSION TITLE
Bookmarklet idea

### DISCUSSION THREAD

**Discussion by petty** [johnste/finicky#251] [General]
I created this Bookmarklet, repeated in each browser I use as ...

**Chrome (default profile):**

Name: !Chrome
Address: `javascript:var%20url=window.location.href;var%20op='?';if(window.location.search){op='&';}url=url+op+'finicky=chrome';url=url.replace(window.location.protocol,'finicky:');window.location.href=url;`

**Chrome (profile 1):**

Name: !Chrome Profile 1 *
Address: `javascript:var%20url=window.location.href;var%20op='?';if(window.location.search){op='&';}url=url+op+'finicky=chrome-profile-1';url=url.replace(window.location.protocol,'finicky:');window.location.href=url;`

\* Finicky supports opening Chrome + a specific Chrome user's profile

**Brave:**

Name: !Brave
Address: `javascript:var%20url=window.location.href;var%20op='?';if(window.location.search){op='&';}url=url+op+'finicky=brave';url=url.replace(window.location.protocol,'finicky:');window.location.href=url;`

**Safari:**

Name: !Safari
Address: `javascript:var%20url=window.location.href;var%20op='?';if(window.location.search){op='&';}url=url+op+'finicky=safari';url=url.replace(window.location.protocol,'finicky:');window.location.href=url;`

... then in my .finicky.js file:
```

    {
      match: ["**finicky=chrome"],
      browser: {
        "name": "Google Chrome",
        "profile": "Default"
      }
    },
    {
      match: ["**finicky=chrome-profile-1"],
      browser: {
        "name": "Google Chrome",
        "profile": "Profile 1"
      }
    },
    {
      match: ["**finicky=brave"],
      browser: "Brave Browser"
    },
    {
      match: ["**finicky=safari"],
      browser: "Safari"
    },
```

Each bookmarklet will convert the page's URL to open with Finicky & process your rules, allowing Finicky to open the URL in an alternate browser.  

So https://github.com/johnste/finicky/ ...
... becomes: finicky://github.com/johnste/finicky/?finicky=brave or safari, etc.

This is in addition to other handlers for email, Slack & so on.

**Comment by petty** [comment]
Also, if you wanted a single "Chooser" popup that presents a list of options you could use this for a single bookmarklet in all browsers:

`javascript:var b=["chrome","chrome-profile-1","safari","firefox","brave"];var url=window.location.href;var w=window.open('','f','width=640,height=240,menubar=0,toolbar=0,location=0,status=0,resizable=0');var p=w.document.body;p.innerHTML='<title>Choose a browser:</title>';url=url+(window.location.search?%27&%27:%27?%27)+%27finicky=%27;url=url.replace(window.location.protocol,%27finicky:%27);b.forEach(el=>{p.innerHTML+=%27<p><a style="font-family:arial;" href="%27+url+el+%27">%27+el+%27</a style=></p>%27});w.focus();`

Note that you would want to update this part of the Bookmarklet for browsers you care about: `var b=["chrome","chrome-bcg","safari","firefox","brave"];` & then update .finicky.js accordingly.

![Screen Shot 2022-03-16 at 8 53 46 AM](https://user-images.githubusercontent.com/94427/158594330-6c3ca464-c509-4fda-b5a3-b5a099a2e002.png)

**Reply by alokdhir** [reply]
Great idea. I did this to make an "open in firefox" bookmark. Shame there's no way to tell Safari to always allow opening Finicky. EDIT: duh the option to Always Allow is right there. Oops. :)
